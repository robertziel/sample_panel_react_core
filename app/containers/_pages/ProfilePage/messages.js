/*
 * ProfilePage Messages
 *
 * This contains all the text for the ProfilePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.ProfilePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Profile page',
  },
  formUsername: {
    id: `${scope}.form.username`,
    defaultMessage: 'Username',
  },
  formEmail: {
    id: `${scope}.form.email`,
    defaultMessage: 'E-mail',
  },
  formPassword: {
    id: `${scope}.form.password`,
    defaultMessage: 'Password',
  },
  formButton: {
    id: `${scope}.form.button`,
    defaultMessage: 'Update',
  },
});
