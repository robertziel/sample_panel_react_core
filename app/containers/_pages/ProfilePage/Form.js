import React, { useState } from 'react';
import { injectIntl, intlShape, FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import {
  Divider,
  Grid,
  SubmitButton,
  TextField,
} from 'components/_ui-elements';

import useApiFetcher from 'containers/BackendApiConnector/fetcher';

import messages from './messages';
import {
  profileUpdateFailedNotify,
  profileUpdateSucceededNotify,
} from './notifications';

function Form({ intl, user }) {
  const fetcher = useApiFetcher();

  // Form state
  const [errorMessages, setErrorMessages] = useState({});
  const [username, setUsername] = useState(user.username);
  const [email, setEmail] = useState(user.email);
  const [password, setPassword] = useState(null);
  const [passwordConfirmation, setPasswordConfirmation] = useState(null);

  const onSubmit = (event) => {
    event.preventDefault();

    fetcher.post({
      disableRetry: true,
      signIn: true,
      path: '/profile',
      body: {
        email,
        password,
        password_confirmation: passwordConfirmation,
        username,
      },
      afterSuccess: (result) => {
        if (result.profile) {
          profileUpdateSucceededNotify();
          setErrorMessages({});
        } else {
          setErrorMessages(result.error_messages);
          profileUpdateFailedNotify();
        }
      },
    });
  };

  const passwordErrorMessage = () => {
    const message = errorMessages.password ? `${errorMessages.password}. ` : '';
    return message + intl.formatMessage(messages.formPasswordLeaveBlank);
  };

  return (
    <form onSubmit={onSubmit}>
      <Grid>
        <TextField
          defaultValue={username}
          label={intl.formatMessage(messages.formUsername)}
          type="text"
          name="username"
          onChange={(event) => setUsername(event.target.value)}
          variant="outlined"
          helperText={errorMessages.username}
          error={!!errorMessages.username}
        />
      </Grid>
      <Grid>
        <TextField
          defaultValue={email}
          label={intl.formatMessage(messages.formEmail)}
          type="email"
          name="email"
          onChange={(event) => setEmail(event.target.value)}
          variant="outlined"
          helperText={errorMessages.email}
          error={!!errorMessages.email}
        />
      </Grid>
      <br />
      <Divider />
      <br />
      <Grid>
        <TextField
          label={intl.formatMessage(messages.formPassword)}
          type="password"
          name="password"
          onChange={(event) => setPassword(event.target.value)}
          variant="outlined"
          helperText={passwordErrorMessage()}
          error={!!errorMessages.password}
        />
      </Grid>
      <Grid>
        <TextField
          label={intl.formatMessage(messages.formPasswordConfirmation)}
          type="password"
          name="password_confirmation"
          onChange={(event) => setPasswordConfirmation(event.target.value)}
          variant="outlined"
          helperText={errorMessages.password_confirmation}
          error={!!errorMessages.password_confirmation}
        />
      </Grid>
      <Grid>
        <SubmitButton processing={fetcher.processing}>
          <FormattedMessage {...messages.formButton} />
        </SubmitButton>
      </Grid>
    </form>
  );
}

Form.propTypes = {
  intl: intlShape.isRequired,
  user: PropTypes.object.isRequired,
};

export default injectIntl(Form);
