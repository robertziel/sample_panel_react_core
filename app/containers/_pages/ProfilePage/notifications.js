import { store } from 'react-notifications-component';

import { getIntl } from 'containers/LanguageProvider/IntlCatcher';
import defaultSettings from 'containers/NotificationsSystem/defaultSettings';

import messages from './messages';

export function profileUpdateSucceededNotify() {
  store.addNotification({
    ...defaultSettings,
    message: getIntl().formatMessage(messages.profileUpdateSucceededNotify),
    type: 'success',
  });
}

export function profileUpdateFailedNotify() {
  store.addNotification({
    ...defaultSettings,
    message: getIntl().formatMessage(messages.profileUpdateFailedNotify),
    type: 'danger',
  });
}
