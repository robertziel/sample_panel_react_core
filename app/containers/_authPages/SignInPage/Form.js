import React, { useState } from 'react';
import useIsMounted from 'react-is-mounted-hook';
import { injectIntl, intlShape, FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { SubmitButton, Note, Grid, TextField } from 'components/_ui-elements';

import { setAuthenticationToken } from 'containers/BackendApiConnector/actions';
import { apiPost } from 'containers/BackendApiConnector/fetchers';

import messages from './messages';
import { signedInNotify } from './notifications';

function Form({ intl, onSignInSuccess }) {
  const isMounted = useIsMounted();
  const [processing, setProcessing] = useState(false);

  // Form state
  const [errorMessage, setErrorMessage] = useState(null);
  const [email, setEmail] = useState(null);
  const [password, setPassword] = useState(null);

  const onSubmit = (event) => {
    event.preventDefault();

    apiPost(
      { isMounted, setProcessing },
      {
        disableRetry: true,
        signIn: true,
        path: '/auth/sign_in',
        body: {
          email,
          password,
        },
        afterSuccess: (result) => {
          setErrorMessage(result.error_message);

          if (result.authentication_token) {
            onSignInSuccess(result.authentication_token);
            signedInNotify();
          }
        },
      },
    );
  };

  return (
    <form onSubmit={onSubmit}>
      <Grid>
        <Note error message={errorMessage} />
      </Grid>
      <Grid>
        <TextField
          label={intl.formatMessage(messages.formEmail)}
          type="email"
          name="email"
          onChange={(event) => setEmail(event.target.value)}
          variant="outlined"
        />
      </Grid>
      <Grid>
        <TextField
          label={intl.formatMessage(messages.formPassword)}
          type="password"
          name="password"
          onChange={(event) => setPassword(event.target.value)}
          variant="outlined"
        />
      </Grid>
      <Grid>
        <SubmitButton processing={processing}>
          <FormattedMessage {...messages.formButton} />
        </SubmitButton>
      </Grid>
    </form>
  );
}

function mapDispatchToProps(dispatch) {
  return {
    onSignInSuccess: (token) => dispatch(setAuthenticationToken(token)),
    dispatch,
  };
}

Form.propTypes = {
  intl: intlShape.isRequired,
  onSignInSuccess: PropTypes.func,
};

export default injectIntl(connect(null, mapDispatchToProps)(Form));
