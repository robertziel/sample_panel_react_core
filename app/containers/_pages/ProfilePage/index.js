/*
 * ProfilePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';

import { Divider, Grid, H1, Paper } from 'components/_ui-elements';
import useApiFetcher from 'containers/BackendApiConnector/fetcher';
import FetchedContent from 'containers/FetchedContent';

import Form from './Form';

import messages from './messages';

export default function ProfilePage() {
  const fetcher = useApiFetcher();

  const [user, setUser] = useState();

  const fetchData = () => {
    fetcher.get({
      path: '/profile',
      afterSuccess: (result) => setUser(result.profile),
    });
  };

  useEffect(() => fetchData(), []);

  return (
    <Grid container>
      <Grid item xs={12}>
        <H1>
          <FormattedMessage {...messages.header} />
        </H1>
        <Divider />
      </Grid>
      <FetchedContent processing={user === undefined || fetcher.processing}>
        <Grid item xs={12} md={12}>
          <Paper>
            <Form user={user} />
          </Paper>
        </Grid>
      </FetchedContent>
    </Grid>
  );
}
