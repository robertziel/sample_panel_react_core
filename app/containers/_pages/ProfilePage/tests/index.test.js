import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';

import { mount } from 'enzyme';
import waitForExpect from 'wait-for-expect';
import { act } from 'react-dom/test-utils';

import loadApiFetchMock from 'testsHelpers/loadApiFetchMock';
import ConfigureTestStore from 'testsHelpers/ConfigureTestStore';

import ProfilePage from '../Loadable';
import Form from '../Form';

const indexPath = '/profile';
const email = 'test@gmail.com';
const username = 'username';
const responseBody = { profile: { email, username } };

let store;
let wrapper;

function mountWrapper() {
  return mount(
    <IntlProvider locale="en">
      <Provider store={store}>
        <ProfilePage />
      </Provider>
    </IntlProvider>,
  );
}

async function configureWrapper() {
  store = new ConfigureTestStore().store;
  await act(async () => {
    wrapper = mountWrapper();
  });
}

describe('<ProfilePage />', () => {
  loadApiFetchMock({
    method: 'GET',
    path: indexPath,
    responseBody,
    status: 200,
  });

  beforeEach(() => {
    configureWrapper();
  });

  it('should render and match the snapshot', async () => {
    await waitForExpect(() => {
      wrapper.update();
      expect(wrapper.find(Form).props().user).toEqual(responseBody.profile);
    });
  });
});
