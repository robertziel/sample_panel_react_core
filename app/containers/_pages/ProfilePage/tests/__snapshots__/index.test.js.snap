// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> should render and match the snapshot 1`] = `"<div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN MuiGrid-container\\"><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN MuiGrid-item MuiGrid-grid-xs-12\\"><h1 class=\\"H1-sc-1hnsqhn-0 eVZYlZ\\"><span>Profile page</span></h1><hr class=\\"MuiDivider-root Divider-l75xrl-0 gGKxTy\\"></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3\\"><div class=\\"MuiPaper-root Paper-t524pg-0 kiWvaW MuiPaper-elevation1 MuiPaper-rounded\\"></div></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9\\"><div class=\\"MuiPaper-root Paper-t524pg-0 kiWvaW MuiPaper-elevation1 MuiPaper-rounded\\"><form><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN\\"></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN\\"><div class=\\"MuiFormControl-root MuiTextField-root TextField-sc-49gm8g-0 ghxyDQ\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled\\" data-shrink=\\"true\\">Username</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" name=\\"username\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"username\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-218 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-220 PrivateNotchedOutline-legendNotched-221\\"><span>Username</span></legend></fieldset></div></div></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN\\"><div class=\\"MuiFormControl-root MuiTextField-root TextField-sc-49gm8g-0 ghxyDQ\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled\\" data-shrink=\\"true\\">E-mail</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" name=\\"email\\" type=\\"email\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"test@gmail.com\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-218 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-220 PrivateNotchedOutline-legendNotched-221\\"><span>E-mail</span></legend></fieldset></div></div></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN\\"><div class=\\"MuiFormControl-root MuiTextField-root TextField-sc-49gm8g-0 ghxyDQ\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\">Password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" name=\\"password\\" type=\\"password\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-218 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-220\\"><span>Password</span></legend></fieldset></div></div></div><div class=\\"MuiGrid-root Grid-sc-1hkmrvu-0 NPciN\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text Button-sc-17yegtu-0 cOSvqW\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\"><span>Update</span></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></form></div></div></div>"`;
